src/post.ts

```diff
@@ -73,28 +79,16 @@ const cleanup = (options: Options) => {
     case 'shell-init':
       return Promise.all([
         removeMicromambaRunShell(options),
-        ...options.initShell.map((shell) => shellDeinit(options, shell))
-      ])
-        .then(() => determineEnvironmentName(options.environmentName, options.environmentFile))
-        .then((environmentName) =>
-          Promise.all(
-            options.initShell.map((shell) => removeEnvironmentFromAutoActivate(options, environmentName, shell))
-          )
-        )
-        .then(() => undefined) // output is not used
+        ...options.initShell.map((shell) => shellDeinit(options, shell)),
+        removeAutoActivation(options)
```

Comment:
hmm... it might be better to not do shellDeinit and and the removal from the rc file synchronously... this might lead to weird race conditions?

Reply:
better safe than sorry ✌️